model_filemanager/download_models.py

```diff
@@ -0,0 +1,78 @@
+import aiohttp
+import os
+from folder_paths import models_dir
+from typing import Callable, Any
+from enum import Enum
+
+from dataclasses import dataclass
+
+class DownloadStatusType(Enum):
+    PENDING = "pending"
+    IN_PROGRESS = "in_progress"
+    COMPLETED = "completed"
+    ERROR = "error"
+
+@dataclass
+class DownloadStatus():
+    status: DownloadStatusType
+    progress_percentage: float
+    message: str
+
+@dataclass
+class DownloadModelResult():
+    status: DownloadStatusType
+    message: str
+    already_existed: bool
+
+async def download_model(session: aiohttp.ClientSession, 
```

Comment:
Is it possible that we break this big function down to smaller pieces for better testability.